File: app/src/main/java/io/github/romangolovanov/apps/ametro/catalog/entities/TransportTypeHelper.java
package io.github.romangolovanov.apps.ametro.catalog.entities;


import java.util.HashSet;
import java.util.Set;

public class TransportTypeHelper {

    public static TransportType parseTransportType(String transportTypeName)
    {
        switch (transportTypeName){
            case "Метро": return TransportType.Subway;
            case "Трамвай": return TransportType.Tram;
            case "Автобус": return TransportType.Bus;
            case "Электричка": return TransportType.Train;
            case "Речной трамвай": return TransportType.WaterBus;
            case "Троллейбус": return TransportType.TrolleyBus;
            case "Фуникулер": return TransportType.CableWay;
            default: return TransportType.Unknown;
        }
    }

    public static TransportType[] parseTransportTypes(String[] names) {
        Set<TransportType> types = new HashSet<>();
        for(String name : names){
            types.add(parseTransportType(name));
        }
        return types.toArray(new TransportType[0]);
    }
}

File: app/src/main/java/io/github/romangolovanov/apps/ametro/catalog/entities/TransportTypeHelper.kt
package io.github.romangolovanov.apps.ametro.catalog.entities

object TransportTypeHelper {

    @JvmStatic
    fun parseTransportType(transportTypeName: String): TransportType = when (transportTypeName) {
        "Метро" -> TransportType.Subway
        "Трамвай" -> TransportType.Tram
        "Автобус" -> TransportType.Bus
        "Электричка" -> TransportType.Train
        "Речной трамвай" -> TransportType.WaterBus
        "Троллейбус" -> TransportType.TrolleyBus
        "Фуникулер" -> TransportType.CableWay
        else -> TransportType.Unknown
    }

    @JvmStatic
    fun parseTransportTypes(names: Array<String>): Array<TransportType> =
        names.mapTo(LinkedHashSet()) { parseTransportType(it) }.toTypedArray()
}
